openlibrary/components/BulkSearch/utils/classes.js

```diff
@@ -122,10 +122,54 @@ export class AiExtractor extends AbstractExtractor{
     }
 }
 
+export class TableExtractor extends AbstractExtractor{
+    /**
+     *
+     * @param {string} name
+     */
+    constructor(name) {
+        super(name)
+        /** @type {boolean} */
+        this.isTable = true
+    }
+
+    /**
+     * @param {ExtractionOptions} extractionOptions
+     * @param {string} text
+     * @return {Promise<BookMatch[]>}
+     */
+    async run(extractionOptions, text){
+
+        /** @type {string[]} */
+        const lines = text.split('\n')
+        /** @type {RegExp} */
+        const matcher = /([\w ]*)\t/g
+        /** @type {string[][]} */
+        const cells = lines.map(textLine => [...textLine.matchAll(matcher)].map(entry => entry[1]))
+        try {
+            /** @type {number} */
+            const authorIndex = cells[0].findIndex(columnName => columnName.trim().toLowerCase()  === extractionOptions.authorColumn)
+            /** @type {number} */
+            const titleIndex = cells[0].findIndex(columnName => columnName.trim().toLowerCase() === extractionOptions.titleColumn)
+            if (titleIndex < 0){
+                throw new Error(`Please have one column named ${extractionOptions.titleColumn} and (optionally) one column named ${extractionOptions.authorColumn}`)
+            }
+
+            return cells.slice(1).filter(row=> row[titleIndex]!== '').map(row => new BookMatch(new ExtractedBook(row[titleIndex], row[authorIndex]), {}))
+        }
+        catch (error){
+            return []
+        }
+    }
+}
 class ExtractionOptions {
     constructor() {
         /** @type {string} */
         this.openaiApiKey = ''
+        /** @type {string} */
+        this.authorColumn = 'author'
```

Comment:
Since this aren't currently configurable, let's simplify the code for now by inlining them an removing them from here.

Reply:
Resolved; the variables are now scoped for the TabularExtractor object, where the definition makes more sense.